prime-router/src/main/kotlin/cli/SenderUtilsCommand.kt

```diff
@@ -5,12 +5,105 @@ import com.github.ajalt.clikt.parameters.options.option
 import com.github.ajalt.clikt.parameters.options.required
 import gov.cdc.prime.router.FileSettings
 import gov.cdc.prime.router.Organization
+import gov.cdc.prime.router.Sender
 import gov.cdc.prime.router.azure.HttpUtilities
 import gov.cdc.prime.router.common.Environment
 import gov.cdc.prime.router.tokens.Scope
 import gov.cdc.prime.router.tokens.SenderUtils
 import java.io.File
 
+// TODO: https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/8659
+// This is a utility created to help migrate keys to organizations and can be removed after the migration is done
+class MigratePublicKeys : SettingCommand(
+    name = "migratekeys",
+    help = """
+        This is a temporary command to assist in the migration of API keys from senders to the organization.
+        
+        It moves all keys from all senders associated with the organization to the organization
+    """.trimIndent()
+) {
+    private val orgName by option(
+        "--orgName",
+        metavar = "<organization>",
+        help = "Specify the name of the organization to add the key to"
+    ).required()
+
+    private val useJson by jsonOption
+
+    val doIt by option(
+        "--doit",
+        help = "Save the modified Organization and senders settings to the database (default is to just print the modified setting)"
+    ).flag(default = false)
+
+    override fun run() {
+        val origOrganizationJson =
+            get(environment, oktaAccessToken, SettingType.ORGANIZATION, orgName)
+        val origOrganization = jsonMapper.readValue(origOrganizationJson, Organization::class.java)
+        val sendersJson = getMany(environment, oktaAccessToken, SettingType.SENDER, origOrganization.name)
+        val orgSenders = jsonMapper.readValue(sendersJson, Array<Sender>::class.java)
+        var updatedOrg = origOrganization
+        val updatedSenders = orgSenders.map(fun(sender): Sender {
+            sender.keys?.forEach(fun(key) {
+                key.keys.forEach(fun(jwk) {
+                    updatedOrg = updatedOrg.makeCopyWithNewScopeAndJwk(key.scope, jwk)
+                })
+            })
+            return sender.makeCopyWithKeyList()
```

Comment:
I don't really like that we had to change the sender classes, especially to add "temporary" code. Can't we manually create a new object here without using the copy constructor and thus changing that code?

Reply:
Yea no objection to that, mostly just following some of the patterns I saw